openpype/hosts/blender/plugins/publish/collect_instances.py

```diff
@@ -19,85 +19,51 @@ class CollectInstances(pyblish.api.ContextPlugin):
+            avalon_prop = obj.get(AVALON_PROPERTY) or {}
             if avalon_prop.get('id') == 'pyblish.avalon.instance':
                 yield obj
 
     @staticmethod
-    def get_collections() -> Generator:
-        """Return all 'model' collections.
-
-        Check if the family is 'model' and if it doesn't have the
-        representation set. If the representation is set, it is a loaded model
-        and we don't want to publish it.
-        """
-        for collection in bpy.data.collections:
-            avalon_prop = collection.get(AVALON_PROPERTY) or dict()
-            if avalon_prop.get('id') == 'pyblish.avalon.instance':
-                yield collection
+    def create_instance(context, group):
+        avalon_prop = group[AVALON_PROPERTY]
+        asset = avalon_prop['asset']
+        family = avalon_prop['family']
+        subset = avalon_prop['subset']
+        task = avalon_prop['task']
+        name = f"{asset}_{subset}"
+        return context.create_instance(
+            name=name,
+            family=family,
+            families=[family],
+            subset=subset,
+            asset=asset,
+            task=task,
+        )
 
     def process(self, context):
         """Collect the models from the current Blender scene."""
         asset_groups = self.get_asset_groups()
-        collections = self.get_collections()
 
         for group in asset_groups:
-            avalon_prop = group[AVALON_PROPERTY]
-            asset = avalon_prop['asset']
-            family = avalon_prop['family']
-            subset = avalon_prop['subset']
-            task = avalon_prop['task']
-            name = f"{asset}_{subset}"
-            instance = context.create_instance(
-                name=name,
-                family=family,
-                families=[family],
-                subset=subset,
-                asset=asset,
-                task=task,
-            )
-            objects = list(group.children)
-            members = set()
-            for obj in objects:
-                objects.extend(list(obj.children))
-                members.add(obj)
-            members.add(group)
-            instance[:] = list(members)
-            self.log.debug(json.dumps(instance.data, indent=4))
-            for obj in instance:
-                self.log.debug(obj)
+            instance = self.create_instance(context, group)
+            members = []
+            if type(group) == bpy.types.Collection:
+                members = list(group.objects)
+                family = instance.data["family"]
+                if family == "animation":
+                    for obj in group.objects:
+                        if obj.type == 'EMPTY' and obj.get(AVALON_PROPERTY):
+                            members.extend(
+                                child for child in obj.children
+                                if child.type == 'ARMATURE')
+            else:
+                members = group.children_recursive
```

Comment:
Wouldn't this case also need the forced `list` conversion?
```suggestion
                members = list(group.children_recursive)
```

Not sure why the others have it but this one seems to be the odd one out. Likely the property is a generator/iterator and using `list` we want to exhaust it directly?

Reply:
Because blender is not very consistent for these things 🥲 `objects` in collections return a `bpy_prop_collection`, while `children` and `children_recursive` in objects are `list`!